app/src/androidTest/java/com/navapp/data/DefaultStoreModelTest.java

```diff
@@ -99,4 +99,37 @@ public void deleteOfTargetDefaultModel_hasDefaultRow_deletesDefault() throws Exc
         DefaultStoreModel storedModel = dao.getDefaultByTable(DefaultStoreTable.RATE);
         assertThat(storedModel, nullValue(DefaultStoreModel.class));
     }
+
+    @Test
+    public void upsertAndRead_doesNotHaveRow_finishesSuccessfully() throws Exception {
+        final DefaultStoreModel INSERT_MODEL = new DefaultStoreModel(
+                DefaultStoreTable.RATE,
+                fakeModels.makeFakeRate().getId()
+        );
+
+        dao.upsert(INSERT_MODEL);
```

Comment:
It's weird that you are inserting stuff into this globul var which is used in other tests, it's of course easier this way, but it could potentially break things.

Reply:
It's a local variable... look few lines above this statement...

Reply:
That's what I meant, my bad, my point stands. It can be manipulated from different test methods.

Reply:
how?

Reply:
Note that each test has a different database, created by `createDb` before the test and closed by `closeDb` after it. `@Before` and `@After` run for each test.

Reply:
Oh, didn't realize it's executed before and after each testing method. My bad.